helpers/wharfapiext/extension.go

```diff
@@ -0,0 +1,181 @@
+package wharfapiext
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"reflect"
+	"regexp"
+
+	"github.com/iver-wharf/wharf-api-client-go/pkg/wharfapi"
+	log "github.com/sirupsen/logrus"
+)
+
+// ExtClient Extension of the wharfapi.Client
+type ExtClient struct {
+	*wharfapi.Client
+}
+
+// CreateOrUpdate Convenience function to POST or PUT data to a Wharf endpoint.
+func (c ExtClient) CreateOrUpdate(data interface{}, method, dataName, endpoint string) (interface{}, error) {
+	var newData = reflect.New(reflect.TypeOf(data)).Elem().Interface()
```

Comment:
Strongly advice against using reflection. Here you're trying to make a generic function in a language that does not support generics. Reflection is forever slow, and the casting required is unsafe and error-prone.

Suggest looking into alternative solutions. Such as just returning the response body bytes as you're doing in your `requests` package instead.